mitmproxy/proxy2/layers/tls.py

```diff
@@ -230,7 +230,7 @@ def _start_tls(self, server: context.Server):
                     server.sni = self.context.client.sni.encode("idna")
                 else:
                     server.sni = server.address[0].encode("idna")
-                    self.tls[server].set_tlsext_host_name(server.sni)
+                self.tls[server].set_tlsext_host_name(server.sni)
```

Comment:
Do you want to cherry-pick these separately and get them merged already or keep it in here? Both is fine with me!